### PR TITLE
react-window: Get rid of unnecessary any types

### PR DESCRIPTION
**Pull request by swissmanu**
* Introduce Generic Type Parameter for itemData
* Change return values of event handlers to void

----

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://react-window.now.sh/#/api/FixedSizeList (and other components)
- [x] Increase the version number in the header if appropriate.

### CODE REVIEW

**Comment by swissmanu** [comment]
will have a look at the failed ci run tomorrow

**Comment by swissmanu** [comment]
all lights green ✅

**Comment by heyimalex** [comment]
Shouldn't the generic default be `undefined`?

It'd be cool if we could make this safer. Because `itemData` is optional, technically everything it gets passed to should be optional too. I took a crack at this a while ago, but couldn't figure out a good way to make it work which is why I settled for `any`. It looks like typescript fixed the issue with generic type inference in react components so maybe it's possible now? Looks something like this:

```ts
type OptionalItemData<T> = T extends undefined
  ? { itemData?: undefined }
  : { itemData: T };

type CommonProps<T = undefined> = OptionalItemData<T> & {
  // The rest of the common props
};
```

Anyhow, I think the odds someone passes `itemData: undefined` to a `FixedSizeList<T>` they manually specified the generic type of is probably really low, so I support this.

**Comment by heyimalex** [comment]
Oh and also there have been changes to the api in 1.6. Check the [changelog](https://github.com/bvaughn/react-window/blob/master/CHANGELOG.md) for details but from [#142](https://github.com/bvaughn/react-window/pull/156) I gather `direction` has basically been renamed to `layout` in lists, and now `direction` is for `"ltr" | "rtl"` rendering.

**Comment by swissmanu** [comment]
thx for your review. will have another look

**Review by heyimalex — CHANGES_REQUESTED** [review]
Just to be clear, my only actual issue with this before merge is that I think the generic default should be `T = undefined`. The other suggestions can be done in separate PRs.

File: types/react-window/index.d.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
-// Type definitions for react-window 1.5
+// Type definitions for react-window 1.6
```
Comment (jgoz) [inline types/react-window/index.d.ts]: You should merge the latest changes from `master`. This has already been bumped to 1.7.

**Review by jgoz — CHANGES_REQUESTED** [review]

**Pull request closed without merging**